projects/observability/src/shared/dashboard/data/graphql/table/entity/entity-table-data-source.model.ts

```diff
@@ -80,12 +110,113 @@ export class EntityTableDataSourceModel extends TableDataSourceModel {
     response: EntitiesResponse,
     request: TableDataRequest<SpecificationBackedTableColumnDef>
   ): TableDataResponse<TableRow> {
+    let results: Entity<string>[] = response.results;
+
+    /**
+     * This block only runs when the table is client side filtered and rendered.
+     * We run an iteration for every filter and find the respective column specification for which filter is to be applied
+     * Then based upon the filter operator, we apply the filter to all the rows
+     */
+    if (
+      this.isClientSideRendered &&
+      request.clientSideFilters?.length !== undefined &&
+      request.clientSideFilters.length > 0
+    ) {
+      request.clientSideFilters.forEach(clientSideFilter => {
+        const matchingColumn = this.findMatchingColumnConfigForClientSideFilter(
+          clientSideFilter.field,
+          request.columns
+        );
+        if (matchingColumn !== undefined) {
+          results = this.filterRespValuesForClientSideFilter(matchingColumn, results, clientSideFilter);
+        }
+      });
+    }
+
+    /**
+     * This block only runs when client side rendering, filtering and sorting is true
+     * For sorting, the column value can either be of Entity type or of Metric Aggregation type.
+     * So, based upon the type and the order, we do the sorting for the respective values
+     */
+    if (request.clientSideSort !== undefined && this.isClientSideRendered && results.length > 0) {
+      if (isMetricAggregation(results[0][request.clientSideSort.column.id])) {
+        results.sort((res1, res2) => {
```

Comment:
Initially read it as `response1`
```suggestion
        results.sort((result1, result2) => {
```

Reply:
Can change, but it's just the scoped var, let me know if it's confusing anyhow.